src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerDriver.cs

```diff
@@ -2776,6 +2776,9 @@ private static bool IsEquivalentSymbol(ISymbol declaredSymbol, ISymbol otherSymb
                                 Debug.Assert(operationBlock.Parent.IsImplicit);
                                 Debug.Assert(operationBlock.Parent.Parent is IConstructorBodyOperation ctorBody &&
                                     ctorBody.Initializer == operationBlock.Parent);
+
```

Comment:
Should we add the below assert?
```suggestion
                                Debug.Assert(!operationBlock.Parent.Parent.IsImplicit);
```